ortools/base/dynamic_library.h

```diff
@@ -91,6 +93,7 @@ class DynamicLibrary {
  private:
   void* library_handle_ = nullptr;
   std::string library_name_;
+  std::vector<std::string> functions_not_found_;
```

Comment:
push_back and vector ... check for efficiency

may be we can reserve to allocate the size of the number of expected functions ?

Reply:
Unfortunately, C++ isn't a reflective language. So if we have e.g

```cpp
struct S {
   int a, b, c;
   double d;
}
```
there is no easy way to count the number of `int` contained in `S`.

Moreover, this is only meant to be executed once (`std::call_once`). And `push_back` is executed only if the XPRESS lib misses some symbols (functions).